### PR TITLE
Support table property override in QueryRewriter

### PR DESCRIPTION
**Pull request by caithagoras**

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryRewriter.java [hunk 1]
```diff
@@ -188,4 +197,18 @@ private static String sanitizeColumnName(String columnName)
...
     }
+
+    private static List<Property> applyPropertyOverride(List<Property> properties, List<Property> overrides)
+    {
+        Map<String, Expression> propertyMap = properties.stream()
+                .collect(toImmutableMap(property -> property.getName().getValue().toLowerCase(ENGLISH), Property::getValue));
+        Map<String, Expression> overrideMap = overrides.stream()
+                .collect(toImmutableMap(property -> property.getName().getValue().toLowerCase(ENGLISH), Property::getValue));
+        return Stream.concat(propertyMap.entrySet().stream(), overrideMap.entrySet().stream())
+                .collect(Collectors.toMap(Entry::getKey, Entry::getValue, (first, second) -> second))
```
Comment (shixuan-fan) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryRewriter.java]: How about naming them `(original, override) -> override`?

File: presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryRewriter.java [hunk 2]
```diff
@@ -47,6 +50,7 @@
 {
     private static final String DEFAULT_PREFIX = "local.tmp";
     private static final QueryConfiguration CONFIGURATION = new QueryConfiguration(CATALOG, SCHEMA, "test_user", Optional.empty(), ImmutableMap.of());
+    private static final List<Property> TABLE_PROPERTIES_OVERRIDE = ImmutableList.of(new Property(new Identifier("retention_days"), new LongLiteral("21")));
```
Comment (shixuan-fan) [inline presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestQueryRewriter.java]: Not sure if it is possible, but maybe we should just use `test_property` for testing?

**Comment by caithagoras** [comment]
@shixuan-fan @mayankgarg1990 comments addressed

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryRewriter.java [hunk 3]
```diff
@@ -38,29 +40,36 @@
...
 public class QueryRewriter
 {
     private final SqlParser sqlParser;
     private final PrestoAction prestoAction;
+    private final List<Property> tablePropertiesOverride;
     private final Map<TargetCluster, QualifiedName> prefixes;
 
     @Inject
-    public QueryRewriter(SqlParser sqlParser, PrestoAction prestoAction, VerifierConfig config)
+    public QueryRewriter(SqlParser sqlParser, PrestoAction prestoAction, List<Property> tablePropertiesOverride, VerifierConfig config)
```
Comment (wenleix) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryRewriter.java]: nit: From english perspective, should this be `tablePropertyOverrides` or `tablePropertiesOverride` ? 

cc @rschlussel

Comment (rschlussel) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryRewriter.java]: I think the first

**Comment by caithagoras** [comment]
@wenleix We need to override the retention of the temporally created tables.

**Pull request merged**